File: Towers.py
import sys


def update(size,floors, position, value):
    position += size

    while (position != 0):
        floors[position] += value
        position /=2

def sum(size, floors, start, end):
    left = start + size
    right = end + size+1

    ret = floors[left]
    while ((left/2) != (right/2)):
        if (left%2 == 0):
            ret += floors[left+1]
        if(right%2 == 1):
            ret += floors[right-1]
        left /=2
        right /=2
    return ret
def showTree(floors):
    for i in floors:
        print floors[i]

while (True):
    numberOfFloors = int(raw_input())

    if (numberOfFloors == 0 ):
        sys.exit()

    size = 1

    while (size <= numberOfFloors):
        size *=2

    floors = [0] * (size*2)

    for i in range(0, numberOfFloors):
        floors[i+size] = int(raw_input())
    for i in range(size-1, 0, -1):
        floors[i] = floors[i*2] + floors[(i*2)+1]
        
    while(True):
        userCommand = sys.stdin.readline().split()

        command = userCommand[0]
        if (command is "S"):
            print "Slutt for i dag."
            break

        a = int(userCommand[1])
        b = int(userCommand[2])
 
        if (command is "I"):
            update(size,floors,a-1,b)
        if (command is "U"):
            update(size,floors,a-1,-b)
        

        if(command is "T"):
            inhabitants = sum(size,floors,a-1,b-1)
            gjest = "gjester"
            if (inhabitants == 1):
                gjest = "gjest"

            if(a == b ):
	        print "Det er %d %s som bor i etasje %d." % (inhabitants,gjest,a)
            elif (a == 1 and  b == numberOfFloors):
	        print "Det er %d %s som bor i hotellet." % (inhabitants,gjest)
            else:
	        print "Det er %d %s som bor mellom etasje %d og etasje %d." % (inhabitants, gjest, a, b)

File: Towers.java
import java.io.*;
import java.util.*;

public class Towers {
    static BufferedReader stdin = new BufferedReader(
            new InputStreamReader(System.in));
    static StringTokenizer st = new StringTokenizer("");
 
    
    public static void main(String[] args) throws Exception {
    	
    	while (true) {
    		int amountOfFloors = readInt();
    		if (amountOfFloors == 0) {
    			break;
    		}
    		int[] floors = new int[amountOfFloors];
    		
    		if (amountOfFloors == 0) {
    			System.exit(0);
    		}
    		for (int i = 0; i < amountOfFloors; i++) {
				floors[i] = readInt();
			}
    		
    		while(true) {
    			int beboere = 0;
    			String command = readString();

    			if (command.equals("S")) {
    				System.out.println("Slutt for i dag.");
    				break;
    			}

    			int a = readInt();
    			int b = readInt();
				}*/

    			
    			if (command.equals("I")) {
    				// Increase
    				floors[a-1] += b;
				}
    			if (command.equals("U")) {
    				// decrease
    				floors[a-1] -= b;
    			}
    			if (command.equals("T")) {
    				// Count
    				for (int i = a-1 ; i < b; i++) {
						beboere += floors[i];
					}
    				if (beboere == 1 && a == b) {
    					System.out.println("Det er 1 gjest som bor i etasje " + a + ".");
    				}
    				else if (beboere == 1) {
    					System.out.println("Det er 1 gjest i som bor mellom etasje " + a + " og etasje " + b  + "." );
    				}
    				else if (a == b) {
    					System.out.println("Det er " + beboere + " gjester som bor i etasje " + a + ".");
    				}
    				
    				else if (a-1 == 0 && b == floors.length) {
    					System.out.printf("Det er %d gjester som bor i hotellet.\n", beboere);
    				}
    				else {
    					System.out.println("Det er " + beboere + " gjester som bor mellom etasje " + a + " og etasje " + b + ".");
    				}
    			}
    		}
    		
    	
    	}

    }

    // Read next input-token as string.
    static String readString() throws Exception {
        while (!st.hasMoreTokens()) {
            st = new StringTokenizer(stdin.readLine());
        }
        return st.nextToken();
    }

    // Read next input-token as integer.
    static int readInt() throws Exception {
        return Integer.parseInt(readString());
    }

    // Read next input-token as double.
    static double readDouble() throws Exception {
        return Double.parseDouble(readString());
    }
}
